spharpy/spherical.py

```diff
@@ -352,40 +379,46 @@ def spherical_harmonic_basis_gradient(n_max, coords):
 
     Examples
     --------
-
     >>> import spharpy
     >>> n_max = 2
-    >>> coords = spharpy.samplings.icosahedron()
-    >>> Y_theta, Y_phi = spharpy.spherical.spherical_harmonic_basis_gradient(
-            n_max, coords)
+    >>> coordinates = spharpy.samplings.icosahedron()
+    >>> grad_theta, grad_phi = spharpy.spherical.spherical_harmonic_basis_gradient(n_max, coordinates)
 
-    """
-    n_points = coords.csize
+
+    """  # noqa: 501
+    if not isinstance(coordinates, pf.Coordinates):
+        axis = np.where(coordinates.shape == 3)[0][0]
+        if axis == 0:
+            coordinates = coordinates.T
+        coordinates = pf.Coordinates(coordinates[:, 0],
+                                     coordinates[:, 1],
+                                     coordinates[:, 2])
```

Comment:
I don't think we should do this transform. We should stick to allowing only pf.Coordinates or Samplings here.

Reply:
It was a leftover from the old branch which I did not touch. But I removed it now.